### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by MuhammadUsmanAslam**
The following changes are being requested to merge:

- Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
   - List all books.
   - List all people.
   - Create a person (teacher or student, not a plain `Person`).
   - Create a book.
   - Create a rental.
   - List all rentals for a given person id.
 - In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
   - Present the user with a list of options to perform.
   - Lets users choose an option.
   - If needed, ask for parameters for the option.
   - Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    end
+  end
+
+  def create_teacher(age, specialization, parent_permission, name)
+    @peoples << Teacher.new(age, specialization, parent_permission: parent_permission, name: name)
+  end
+
+  def create_student(age, name, parent_permission)
+    @peoples << Student.new(age, name, parent_permission: parent_permission)
+  end
```
Comment (danmainah) [inline app.rb]: - [x] @MuhammadUsmanAslam great work on your app, however, am unable to create a new person either a teacher or student am getting the errors below 
image
image

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
+require_relative './app'
+
+def run_option
+  puts 'Welcome to the School Library App!'
```
Comment (danmainah) [inline main.rb]: - [x] @MuhammadUsmanAslam Great work on your interface, however, I would recommend you ensure the `Welcome to the Schoo Library App!` only appears when the app is launched and not whenever a person choose the options in the menu as shown below 
image

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    @books << Book.new(title, author)
+  end
```
Comment (danmainah) [inline app.rb]: - [x] @MuhammadUsmanAslam Great work on your app. However am unable to add a new book, am getting the error below 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @MuhammadUsmanAslam , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights


# Required Changes ♻️

- [x] @MuhammadUsmanAslam Kindly ensure all the app functions are working as expected from person book and rental creation and also viewing the same.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title} Author: #{book.author} \n" }
+  end
+
+  def list_peoples
+    @peoples.each do |people|
+      puts "[#{people.class.name}] Name: #{people.name} ID: #{people.id} Age: #{people.age} \n"
+    end
+  end
```
Comment (AmaduKamara) [inline app.rb:22]: - [x] When listing people, the student's name is expected to show but it's showing unknown. Could you please check that out? Only the teacher's name is displayed when listing people. 👍🏽

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  def create_teacher(age, specialization, name, parent_permission)
+    @peoples << Teacher.new(age, specialization, parent_permission: parent_permission, name: name)
+  end
```
Comment (AmaduKamara) [inline app.rb]: - [x] According to the requirement, when creating a teacher, it's not required to ask for parent permission for a teacher. Only when creating a student are you asked for parent permission for that student. Could you please work on that? Please see this [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) for better understanding 👍🏽
- [x] Also, when listing people, the student's name is not displayed. Please work on that 👍🏽

Screenshot 2023-05-26 at 11 56 20 AM

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number(not ID)'
+    @peoples.each.with_index do |person, idx|
+      puts "#{idx}) [#{person.class.name}] Name #{person.name}, ID #{person.id}, Age #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Date:-'
+    date = gets.chomp
+    Rental.new(@books[book_index], @peoples[person_index], date)
+  end
```
Comment (AmaduKamara) [inline app.rb]: - [ ] I can't create a rental for a person at the moment. When doing so, the app crashes with the error - undefined method rentals. Kindly check it out 👍🏽

Screenshot 2023-05-26 at 12 00 19 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @MuhammadUsmanAslam,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Created UI for the school library app 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please work on the display of the student's name and the creation of a teacher without asking for parental permission for a teacher. 👍🏽
- [ ] Rental creation is not working. Please work on that 👍🏽
- [ ] Kindly check all features are working as expected before you submit for a review 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp.to_s
+    puts 'Author:'
+    author = gets.chomp.to_s
+    @books << Book.new(title, author)
+  end
+
+  def create_rental
```
Comment (1GORDON) [inline app.rb:40]: Great job @MuhammadUsmanAslam 🤝

- [x] Great job on implementing the changes requested by the previous reviewer. Kindly ensure the rentals are created successfully and can be displayed in the UI. At the moment there is no error but the created rentals are not visible in the UI.
rental

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @MuhammadUsmanAslam ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @MuhammadUsmanAslam , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @MuhammadUsmanAslam , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @MuhammadUsmanAslam , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**